test/se/vidstige/jadb/test/unit/MockedTestCases.java

```diff
@@ -104,11 +104,17 @@ public void testExecuteShell() throws Exception {
     }
 
     @Test
-    public void testExecuteShellQuotesSpace() throws Exception {
+    public void testExecuteShellQuotesWhitespace() throws Exception {
         server.add("serial-123");
         server.expectShell("serial-123", "ls 'space file'").returns("space file");
+        server.expectShell("serial-123", "echo 'tab\tstring'").returns("tab\tstring");
+        server.expectShell("serial-123", "echo 'newline1\nstring'").returns("newline1\nstring");
+        server.expectShell("serial-123", "echo 'newline2\r\nstring'").returns("newline2\r\nstring");
         JadbDevice device = connection.getDevices().get(0);
         device.executeShell("ls", "space file");
+        device.executeShell("echo", "tab\tstring");
+        device.executeShell("echo", "newline1\nstring");
+        device.executeShell("echo", "newline2\r\nstring");
```

Comment:
ℹ️  Normally one case per test is good, e.g. tab and newline has their own test functions. 
ℹ️  Also the stdout of the expected calls are not used so these could be dropped, e.g. `server.expectShell("serial-123", "echo 'newline2\r\nstring'").returns("newline2\r\nstring");` could be just `server.expectShell("serial-123", "echo 'newline2\r\nstring'");`